persistence/gov.go

```diff
@@ -16,15 +20,33 @@ const (
 	ServiceNodesPerSessionParamName = "service_nodes_per_session"
 )
 
-// TODO (Team) BUG setting parameters twice on the same height causes issues. We need to move the schema away from 'end_height' and
-// more towards the height_constraint architecture
+// Mapping of parameter names and their stringified type names
+var parameterNameTypeMap = make(map[string]string)
 
-func (p PostgresContext) GetBlocksPerSession(height int64) (int, error) {
-	return p.GetIntParam(BlocksPerSessionParamName, height)
-}
-
-func (p PostgresContext) GetServiceNodesPerSessionAt(height int64) (int, error) {
-	return p.GetIntParam(ServiceNodesPerSessionParamName, height)
+// Using the json tag of the fields in the Params struct to extract the name of the
+// parameters and build a mapping in memory of the types associated to each parameter
+// name according to the struct generated from: persistence_genesis.proto
+func init() {
+	st := reflect.TypeOf(genesis.Params{})
+	ownerRegEx, err := regexp.Compile("_owner$")
+	if err != nil {
+		log.Fatalf("error compiling regular expression `_owner$` to match paramNames.")
+	}
+	// Loop through struct fields to build ParameterNameTypeMap
+	for i := 0; i < st.NumField(); i++ {
+		field := st.Field(i)
+		json := field.Tag.Get("json") // Match the json tag of field: json:"paramName,omitempty"
+		paramName := strings.Split(json, ",")[0]
+		typ := field.Type.Name() // Get string version of field's type
+		matchOwner := ownerRegEx.MatchString(paramName)
+		// Handle string types
+		if typ == "string" && matchOwner {
+			typ = "[]uint8" // All owners are string types but used as []byte slices
+		} else if typ == "string" {
+			typ = "*big.Int" // Rest of string types are used as *big.Int
+		}
```

Comment:
I'd like to remove lines 42-47 altogether. It might be the case right now, but it means we can never have a non integer string parameter. 

Does anything fail if we remove this?

Reply:
@Olshansk No nothing will fail, I have changed 1 unit test back to using `GetBytesParam` as removing the `_owner` check on the param name means again that there will not be any `[]byte` types, as none of the Params have this type. I will leave the `[]byte` case in the switch though so it doesn't need to be added at another time. I am imagining a follow up to this task would include sorting out this parameter type issue so better to have the 3 getter functions covered.

Reply:
`I am imagining a follow up to this task would include sorting out this parameter type issue so better to have the 3 getter functions covered.`

I think that's a good idea and I like the simplification right now.